File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        n = len(s)
        l = 0
        r = n-1
        count = 0
        def vp(s,l,r,count):
            if count == 2:
                return False
            if l>=r:
                return True
            
            if s[l]==s[r]:
                return vp(s,l+1,r-1,count)
            else:
                return vp(s,l+1,r,count+1) or vp(s,l,r-1,count+1)
        
        return vp(s,l,r,0)
        

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        
        if(root==null){
            return 0;
        }
        int num = root.val;
        return 1+ count(root.left, num) + count(root.right, num);

    }

    private int count(TreeNode root, int num){
        if (root == null){
            return 0;
        }

        if (root.val >= num){
            return 1 + count(root.left,root.val) + count(root.right,root.val);
        }
        else{
            return count(root.left,num) + count(root.right,num);
        }
    }
}
